src/vega_viewer/vega/specs/grouped_stacked_chart.cljs

```diff
@@ -0,0 +1,186 @@
+                                        :y {:scale "y"
+                                            :field "sum_y_start"}
+                                        :y2 {:scale "y"
+                                             :field "sum_y_end"}
+                                        :fill {:scale "color"
+                                               :field "z"}}}}]
+                     :axes [{:type "y"
+                             :scale "y"
+                             :grid true
+                             :tickSize 0
+                             :properties
+                             {:labels {:text {:value ""}}
+                              :axis {:stroke {:value "transparent"}}}
+                             :layer "back"}]}]
+            :scales [{:name "column"
+                      :type "ordinal"
+                      :domain {:data "summary"
+                               :field "u"
+                               :sort true}
+                      :range "width"
+                      :round true}
+                     {:name "x"
+                      :type "ordinal"
+                      :domain {:data "summary"
+                               :field "x"
+                               :sort true}
+                      :bandSize 21
+                      :round true
+                      :points true
+                      :padding 1}
+                     {:name "y"
+                      :type "linear"
+                      :domain {:data "stacked_scale"
+                               :field "sum_sum_y"}
+                      :rangeMin 200
+                      :rangeMax 0
+                      :round true
+                      :nice true
+                      :zero true}
+                     {:name "color"
+                      :type "ordinal"
+                      :domain {:data "summary"
+                               :field "z"
+                               :sort true}
+                      :range "category20"}]
+            :axes [{:type "x"
+                    :scale "column"
+                    :orient "top"
+                    :tickSize 0
+                    :properties
+                    {:axis {:strokeWidth {:value 0}}}}]
+            :legends [{:fill "color"
+                       :properties
+                       {:symbols {:shape {:value "circle"}
+                                  :strokeWidth {:value 0}}}}]}]})
+
+(defn generate-grouped-stacked-chart-vega-spec
+  [{:keys [data height width status-text
+           maximum-y-axis-label-length]}
+   & {:keys [responsive? user-defined-palette]}]
+  (let [chart-height (min (or height (* (count data) band-width)) max-height)
+        chart-width (or width
+                        (and (not responsive?)
+                             default-chart-width))]
+    (-> grouped-stacked-chart-spec-template
+        (assoc-in [:data 0 :values] data)
+        (assoc :height chart-height)
```

Comment:
this can be combined with the below line, https://clojuredocs.org/clojure.core/assoc#example-542692c9c026201cdc326acd